gtda/local_homology/simplicial.py

```diff
@@ -0,0 +1,411 @@
+from numbers import Real
+from types import FunctionType
+
+import numpy as np
+import warnings
+
+from scipy.spatial.distance import pdist, squareform
+from sklearn.neighbors import KNeighborsTransformer, RadiusNeighborsTransformer
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+from gtda.homology import VietorisRipsPersistence
+from gtda.utils.intervals import Interval
+from gtda.utils.validation import validate_params
+from gtda.plotting import plot_diagram
+from gtda.base import PlotterMixin
+from gtda.utils._docs import adapt_fit_transform_docs
+
+
+class LocalVietorisRipsBase(BaseEstimator,
+                            TransformerMixin,
+                            PlotterMixin):
+    """
+    Base class for KNeighboursLocalVietorisRips and RadiusLocalVietorisRips.
+    LocalVietorisRipsBase is not meant to be used. Please see documentation
+    for KNeighboursLocalVietorisRips and RadiusLocalVietorisRips."""
+
+    def __init__(self, metric='euclidean', homology_dimensions=(1, 2),
+                 neighborhoud_param=(1, 2), n_jobs=1):
+        """
+        Initializes the base class by setting the parameters, initializes the
+        object used for computing persistence homology, and checking that the
+        parameters were initialized correctly."""
+
+        # metric for the point cloud
+        self.metric = metric
+
+        # topological dimension of features to be computed
+        self.homology_dimensions = homology_dimensions
+
+        self.n_jobs = n_jobs
+
+        # Object is used to compute persistence diagrams
+        self.homology = VietorisRipsPersistence(
+                            metric='precomputed',
+                            collapse_edges=True,
```

Comment:
Have you compared runtime between `collapse_edge=True` and `collapse_edges=False`?  Would it make sense to let the user decide this?  Collapser is risky: if you have small local point clouds (which I think is the case here) and are not pushing the computation beyond homology dimension 1 it can actually make things slower.

Reply:
I haven't tested it no, but if what you are saying is true then there are definitely cases where it is slower.  Definitely, letting the user decide is an option. I'll add it as an additional argument.

Reply:
Thank you!  Yes, False default please. You should be able to copy the docstring from the `VietorisRipsPersistence`

Reply:
Done! It made the tests go slower, since the dimensions there are fairly high. I changed half the tests to run with collapse_edges=True. Is this good, or should I make all of them run with collapse_edges=True in order to improve runtime?

Reply:
Thanks!  I think leaving `True` in all tests is good, since we are confident that edge collapser gives the correct results.